Changed D2k health, armor types and weapon stats to match original

reaperrr opened the pull request:
A large chunk from #9106.

reaperrr committed:
Changed d2k infantry HP to their original values

reaperrr committed:
Changed d2k structure HP and armor types to their original values

reaperrr committed:
Changed d2k vehicle HP and armor types to their original values

reaperrr committed:
D2k weapon changes to reload, burst and range

reaperrr committed:
D2k weapon projectile changes

speed, trails, rate of turn, projectile type etc.

reaperrr committed:
D2k weapon warhead changes

reaperrr committed:
D2k weapon targeting validity changes

reaperrr committed:
D2k added Grenadier and Sardaukar explosions

reaperrr committed:
Missile tank weapon renamed

GraionDilach commented on mods/d2k/weapons.yaml:
```diff
@@ -83,68 +106,92 @@ M_LMG:
-	BurstDelay: 5
+	ReloadDelay: 20
+	Range: 3c0
+#	Burst: 3	# Actually fires 3 times in D2k but deals damage all at once, purely visual/audio effect
```
Whatever uses this could have the weapon defined three times on it instead then with 1/3 damage.

reaperrr replied:
That wouldn't reproduce the original behavior either.

I opted fro `Burst: 3` and `BurstDelay: 2` with `Damage: 60` per burst instead. That looks and sounds pretty much like the original.

Mailaender commented:
You need to remove that `[PaletteReference]` from https://github.com/reaperrr/OpenRA/blob/d2k-bal-weapons/OpenRA.Mods.Common/Effects/Missile.cs#L62 as that can be a player palette reference which is not yet supported. https://github.com/OpenRA/OpenRA/issues/9101

reaperrr committed:
Temporary work-around for D2k deviator missile

The lint check doesn't handle player palettes properly yet.

reaperrr committed:
Split Trike and Raider muzzle flash to its own weapon

Allows to show 3 muzzle flashes while only playing report, explosion effect and applying damage once

reaperrr commented:
Updated.

Additionally improved weapons after looking at D2k original weapon stats myself:
- split Trike/Raider muzzle to separate weapon (for 3 muzzle flashes)
- reduced raider weapon range from 5 to 3 cells to match trike and original
- removed redundant contrail definitions from various MGs
- removed explosion sound from tank cannons, less annoying and matches original
- changed some other firing and explosion sounds and effects to be less annoying and better match original
- tweaked projectile speeds, should match original even more closely now
- reduced devastator weapon range from 4 to 3 cells to match original
- added projectile shadow to grenadier's grenade

OmegaBolt commented:
DevBullet has range 128 in D2k which should be 4 cells.

reaperrr commented:
> DevBullet has range 128 in D2k which should be 4 cells.

Which version of D2k? I looked at both the Gruntmod edition advertised by CnCNet as well as a seemingly vanilla edition from a abandonware site, and in both cases TibEd claims a range of 96.

OmegaBolt commented:
Hmm, mine should just be the original release with 1.06 patch.

penev92 commented:
I currently have the Gruntsmods version installed and TibEd says 128:

![](http://i.imgur.com/C6o3qcV.png)

reaperrr commented:
Then I don't know what's going on here. I've tried reinstalling TibEd, I've tried reinstalling Gruntmod, I've unzipped vanilla 1.06 and used the bins from that - same result all the time, TibEd stubbornly shows 96 for me.

The only explanation I can think of is that TibEd somehow somewhere stored some kind of temporary file with these values, and now reloads that all the time instead of directly reading the bin files.

OmegaBolt commented:
AFAIK TibEd does just that, under Data\bin in the D2k folder.

reaperrr commented:
I've overwritten all .bin files with the originals from the official 1.06 patch, no change.

penev92 commented:
Well I have no idea what's going on, but let's not get hung up on that for the next 3-5 weeks. I say pick one (the higher, as the tank is slow enough already) and go with it.

reaperrr commented:
Done.

reaperrr commented:
I wanted to know for certain and did some ingame testing. It appears that the range was 3 cells in the original release, but later increased to 4 in one of the official patches (it's definitely 4 with 1.06). So it seems my TibEd just doesn't refresh some cached(?) value. 4 cells is correct in any case.

reaperrr committed:
Fix Grenadier projectile

reaperrr commented:
Updated. Fixed:
- Grenadier projectile image
- Bazooka trail and Quad rocket explosion
- Siege Tank 155mm explosion

Mailaender commented:
The death hand missile is not doing any damage anymore and the repair pad doesn't allow vehicle repairs (might have regressed earlier). I also noticed that weapon ranges are extremely short now. Not sure if that really was original game balancing or is a conversion error.

reaperrr commented:
They really were that short in the original. However, our MoveIntoRange logic probably moves units 1 cell closer to target than the original did, so it might feel even shorter.

reaperrr commented:
Updated death hand damage. Should now kill most buildings when they are hit directly unless they have full health or are conyards.

Mailaender commented on mods/d2k/sequences/misc.yaml:
```diff
@@ -261,6 +266,11 @@ bombs:
 		Length: 4
 
+grenade:
+	idle: grenade.shp
```
Bonus points if you document which frame number you extracted those.

Mailaender commented on OpenRA.Mods.Common/Effects/Missile.cs:
```diff
@@ -59,7 +59,8 @@ class MissileInfo : IProjectileInfo
 
-		[PaletteReference] public readonly string TrailPalette = "effect";
+		// TODO: Re-add PaletteReference as soon as it handles player palettes
+		public readonly string TrailPalette = "effect";
```
I just filed https://github.com/OpenRA/OpenRA/pull/9384 so if you want to wait.

pchote commented:
From a technical point this gets my :+1: with or without the player palette lint pr.  I haven't made any real effort to verify that the stats match the original game, except to check that the trooper seems consistently OP in both :wink:

penev92 commented on mods/d2k/rules/structures.yaml:
```diff
@@ -49,9 +49,9 @@ construction_yard:
+		HP: 3000
 	Armor:
-		Type: Concrete
+		Type: cy
```
TibEd says this uses "LIGHT" ?

penev92 replied:
OK looking at the "Damage versus" section of the rules (in TibEd), that doesn't make any sense. Also TibEd does say that "CY" stands for "Construction Yard". So I guess it's quite possible the "LIGHT" value is an error.

penev92 commented:
Looks fine to me :+1:

penev92 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/711b5c646f895921c629fc47a6e4bb3d26b1718e...fc04f185052181dbe82f062970606394d718e4c0)
